### DISCUSSION TITLE
HSS Payload Question

### DISCUSSION THREAD

**Discussion by erentkll** [polarfire-soc/polarfire-soc-documentation#323] [HSS and U-Boot] [answered]
Hi everyone, I am new to PolarFire SoC. I have a FreeRTOS+POSIX project and I want to run this application on system boot. I suppose I need to use HSS and generate payload from my project binary file. To learn this, I have watched the video in the link https://youtu.be/LseJta8qVbM 
 However, in the video, hss-payload-generator.exe is used and it does not mention about how to create this .exe application. In the current version of hart-software-services-master, there is no such application. My question is how can I run my FreeRTOS+POSIX application on system boot, also if I need to generate payload for HSS, how to generate this payload? Thank you in advance.
Regards,

**Comment by hughbreslin** [comment, marked as answer]
Hey @erentkll 

the sources for the payload generator are bundled in the tools folder of the HSS [here](https://github.com/polarfire-soc/hart-software-services/tree/master/tools/hss-payload-generator). 

A prebuilt executable is also available as an asset with the 2023.02 release [here](https://github.com/polarfire-soc/hart-software-services/releases/tag/v2023.02).

The [readme](https://github.com/polarfire-soc/hart-software-services/tree/master/tools/hss-payload-generator#introduction) for the payload generator has examples of how to use it and there are also some sample configuration files in the test directory. 

Running your application depends on where you are running it, you can create a payload to have it run on boot from target memory and then attach to debug it. You can also use the HSS to configure the system and then interrupt it before it loads a payload (or disable ‘CONFIG_SERVICE_BOOT’ so it doesn’t try and load a payload) and then load your application via the debugger.

**Reply by erentkll** [reply]
Thank you so much for your answer, I can manage to generate output.bin file now. I have one more question with apologies for taking your time. My question is to write the application to eMMC on HSS, can I use this .bin file with win32 disk imager or should I need an image file? If so, how can I generate the image file? Thank you so much again you are very helpful.

**Reply by hughbreslin** [reply]
No problem :)

You can write the .bin file using win32 disk imager - you might need to set it to look for all file types when you are trying to select the .bin
